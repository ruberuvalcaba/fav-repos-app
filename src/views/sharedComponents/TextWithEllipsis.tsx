import React from 'react'
import { Text } from '@chakra-ui/react'

const TextWithEllipsis = ({
  children,
  width,
}: {
  children: React.ReactNode
  width: string
}): JSX.Element => {
  return (
    <Text
      fontSize="xs"
      style={{
        display: 'block',
        width: width,
        whiteSpace: 'nowrap',
        overflow: 'hidden',
        textOverflow: 'ellipsis',
      }}
      _hover={{
        overflow: 'visible',
      }}
    >
      <>{children}</>
    </Text>
  )
}

export default TextWithEllipsis
