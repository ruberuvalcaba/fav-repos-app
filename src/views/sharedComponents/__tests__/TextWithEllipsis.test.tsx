import React from 'react'
import renderer from 'react-test-renderer'
import TextWithEllipsis from '../TextWithEllipsis'

it('TextWithEllipsis component renders correctly', () => {
  const component = renderer.create(
    <TextWithEllipsis width="100px">This is a text example with ellipsis</TextWithEllipsis>
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
