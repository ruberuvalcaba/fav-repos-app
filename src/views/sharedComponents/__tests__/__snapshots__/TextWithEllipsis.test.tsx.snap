// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextWithEllipsis component renders correctly 1`] = `
[
  <style
    dangerouslySetInnerHTML={
      {
        "__html": ".css-1wiw5gj{font-size:xs;}.css-1wiw5gj:hover,.css-1wiw5gj[data-hover]{overflow:visible;}",
      }
    }
    data-emotion="css 1wiw5gj"
  />,
  <p
    className="chakra-text css-1wiw5gj"
    style={
      {
        "display": "block",
        "overflow": "hidden",
        "textOverflow": "ellipsis",
        "whiteSpace": "nowrap",
        "width": "100px",
      }
    }
  >
    This is a text example with ellipsis
  </p>,
]
`;
